code/game/objects/structures/nest.dm

```diff
@@ -0,0 +1,65 @@
+// Nests spawn mobs and objects when crossed by players.
+// Nests placed in "spawn_trigger_distance" will always spawn the same type of mob.
+
+/obj/structure/nest
+	name = "nest"
+	desc = "A twisted, dark passage to the underground."
+	icon = 'icons/mob/nest.dmi'
+	icon_state = "hole"
+
+	move_resist = INFINITY
+	anchored = TRUE
+	density = FALSE
+
+	var/faction = list("hostile")
+	var/spawn_byproduct = list(/obj/item/stack/ore/glass, /obj/item/stack/ore/iron)
+	var/spawn_byproduct_max = 3
+	var/spawn_max = 2
+	var/spawn_mob_options = list(/mob/living/simple_animal/crab)
+	var/spawn_trigger_distance = 7
+
+/obj/structure/nest/Initialize(mapload)
+	..()
+	RegisterSignal(src, COMSIG_MOVABLE_CROSSED, .proc/try_spawn)
```

Comment:
You can just override `Crossed`. Don't have to use a signal here as it overcomplicates what you're trying to do.

Reply:
If I simply override crossed, then people can criss-cross nests over and over again. Furthermore, if they cross a nest once, then run to a nearby nest, they can cross that as well, and instead of 3 spawns, we would get 6!

Reply:
Having a `has_spawned` variable would solve this. It would also allow you to easily override `examine` to do the change AA suggested.

Reply:
Removed the signal, overrode `examine()` and `Crossed()` instead as requested! In 587c6161feb3fe89cbc5d2686ffebe15c72966df